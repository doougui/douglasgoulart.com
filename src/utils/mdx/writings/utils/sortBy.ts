import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
