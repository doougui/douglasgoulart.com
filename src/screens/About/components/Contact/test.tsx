import { render, screen } from 'utils/tests';

import { Contact } from '.';

describe('<Contact />', () => {
  it('should render title and contact button', () => {
    const { container } = render(<Contact />);

    expect(
      screen.getByRole('heading', { name: /Do you like what you see?/i }),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('link', { name: 'Send me a message' }),
    ).toHaveAttribute('href', 'mailto:douglaspigoulart@gmail.com');

    expect(container.firstChild).toMatchSnapshot();
  });
});
