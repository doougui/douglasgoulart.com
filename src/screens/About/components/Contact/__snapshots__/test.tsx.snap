// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render title and contact button 1`] = `
.c2 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 1.6rem;
  line-height: 2.4rem;
  color: #EDEDEF;
  background-color: #107BEB;
  border-radius: 0.4rem;
  padding: 1rem 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  font-size: 2.4rem;
  line-height: 3rem;
  color: #EDEDEF;
  font-weight: 600;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2rem;
}

<section
  class="c0"
  spacing="2rem"
>
  <h2
    class="c1"
  >
    Do you like what you see?
  </h2>
  <a
    class="c2"
    href="mailto:douglaspigoulart@gmail.com"
  >
    Send me a message
  </a>
</section>
`;
